Sources/FoundationEssentials/String/StringProtocol+Essentials.swift

```diff
@@ -255,8 +260,12 @@ extension String {
             // Other encodings, defer to the CoreFoundation implementation
             return _ns.data(using: encoding.rawValue, allowLossyConversion: allowLossyConversion)
 #else
-            // Attempt an up-call into swift-corelibs-foundation, which can defer to the CoreFoundation implementation
-            return _cfStringEncodingConvert(string: self, using: encoding.rawValue, allowLossyConversion: allowLossyConversion)
+            return (
+                // Attempt an up-call into swift-corelibs-foundation, which can defer to the CoreFoundation implementation
+                _cfStringEncodingConvert(string: self, using: encoding.rawValue, allowLossyConversion: allowLossyConversion) ??
+                // Or attempt an up-call into ICU via FoundationInternationalization
+                _icuStringEncodingConvert(string: self, using: encoding, allowLossyConversion: allowLossyConversion)
```

Comment:
When do we fall back to the ICU one here? On Darwin this encoding is handled via CF, which itself calls into ICU.

Reply:
I guess that'll happen if `import Foundation` has not been done (and CF is present), but `import FoundationInternationalization` has. I wonder how we can further unify these paths; on Darwin too...

Reply:
I had a similar question - I believe `_cfStringEncodingConvert` will return `nil` if any of the following are the case:

1. CF isn't loaded
2. CF is loaded but it doesn't support the provided encoding
3. CF is loaded and supports the encoding but the data is malformed

We want the up-call in 2, and I think Tony's question was about 1, but do we want the up call in case 3?

Reply:
I now agree that unexpected fallback should be avoided.
Although ICU *can* support other encodings, I shoud've make this change focus on EUC-JP.